src/contracts/core/Slasher.sol

```diff
@@ -83,534 +26,45 @@ contract Slasher is Initializable, OwnableUpgradeable, ISlasher, Pausable {
-     * slash `operator` once `serveUntilBlock` is reached
-     */
-    function recordLastStakeUpdateAndRevokeSlashingAbility(
-        address operator,
-        uint32 serveUntilBlock
-    ) external onlyRegisteredForService(operator) {
-        // update the 'stalest' stakes update time + latest 'serveUntilBlock' of the `operator`
-        _recordUpdateAndAddToMiddlewareTimes(operator, uint32(block.number), serveUntilBlock);
-        // remove the middleware from the list
-        require(
-            _operatorToWhitelistedContractsByUpdate[operator].remove(_addressToUint(msg.sender)) != 0,
-            "Slasher.recordLastStakeUpdateAndRevokeSlashingAbility: Removing middleware unsuccessful"
-        );
-        // revoke the middleware's ability to slash `operator` after `serverUntil`
-        _revokeSlashingAbility(operator, msg.sender, serveUntilBlock);
-    }
-
-    // VIEW FUNCTIONS
-
-    /// @notice Returns the block until which `serviceContract` is allowed to slash the `operator`.
-    function contractCanSlashOperatorUntilBlock(
-        address operator,
-        address serviceContract
-    ) external view returns (uint32) {
-        return _whitelistedContractDetails[operator][serviceContract].contractCanSlashOperatorUntilBlock;
-    }
-
-    /// @notice Returns the block at which the `serviceContract` last updated its view of the `operator`'s stake
-    function latestUpdateBlock(address operator, address serviceContract) external view returns (uint32) {
-        return _whitelistedContractDetails[operator][serviceContract].latestUpdateBlock;
-    }
-
-    /*
-     * @notice Returns `_whitelistedContractDetails[operator][serviceContract]`.
-     * @dev A getter function like this appears to be necessary for returning a struct from storage in struct form, rather than as a tuple.
+     * @notice External function called by AVSs to register an operator without the ability to slash it
+     * @param operator The address of the registering operator
+     * @param expiry The desired expiry time of the operator's signature
+     * @param signature The signature of the operator
      */
-    function whitelistedContractDetails(
-        address operator,
-        address serviceContract
-    ) external view returns (MiddlewareDetails memory) {
-        return _whitelistedContractDetails[operator][serviceContract];
+    function registerOperatorWithoutSlashingWithAVS(address operator, uint256 expiry, bytes memory signature) external {
+        require(expiry > block.timestamp, "Slasher.registerOperatorWithoutSlashingWithAVS: signature expired");
+        require(!registeredWithoutSlashing[msg.sender][operator], "Slasher.registerOperatorWithoutSlashingWithAVS: operator already registered without slashing");
+        // check operator signature
+        EIP1271SignatureUtils.checkSignature_EIP1271(operator, calculateRegistrationWithoutSlashingDigestHash(operator, msg.sender, nonces[msg.sender]++, expiry), signature);
```

Comment:
I think it would make more sense to use a 'salt' here instead of an incrementing nonce

Reply:
salts feel cleaner fs

Reply:
> sure, what's your deciding criteria though?

my thought process is simply "do these things need to be in order? (if yes, then use a nonce), or will enforcing order lead to undesired race conditions? (if yes, then use a salt)"